### PR TITLE
70B training data generation.

### PR DESCRIPTION
**Pull request by ita9naiwa**

### CODE REVIEW

**Review by adstraw — COMMENTED** [review]
Please add top-level README to document all commands to 1) generate data 2) fine tune and 3) evaluate fine tune for both llama3.1-8b and llama3.1-70b.

File: eagle/ge_data/ge_data_all_llama3-1-70Bchat.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+# os.environ["CUDA_VISIBLE_DEVICES"] = str(args.gpu_index)[1:-1]
+import torch
+import torch.nn.functional as F
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer,BitsAndBytesConfig
+from datasets import load_dataset
+import json
+from fastchat.model.model_adapter import get_conversation_template
+
+bigname="/opt/models/meta-llama-3.1-70b-instruct"
```
Comment (adstraw) [inline eagle/ge_data/ge_data_all_llama3-1-70Bchat.py]: add --base-model-path arg

File: eagle/ge_data/ge_data_all_llama3-1chat.py [hunk 2]
```diff
@@ -46,12 +49,9 @@ def build_dataset_rank(
     ds = load_dataset('json', data_files="ShareGPT_V4.3_unfiltered_cleaned_split.json")
     ds = ds['train']
     ds = ds.shuffle(seed=42)
-    ds1 = ds.select(range(args.start, args.end))
-    # ds1 = ds.select(range(100,200))
-    # dst=ds.select(range(200,300))
-    # ds2=ds.select(range(300,len(ds)))
+    # ds1 = ds.select(range(args.start, args.end))
+    ds1 = ds.select(range(0, 1000))
```
Comment (adstraw) [inline eagle/ge_data/ge_data_all_llama3-1chat.py]: Do we want this hard-coded?

File: eagle/ge_data/ge_data_all_llama3-1-70Bchat.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+# bigmodel = AutoModelForCausalLM.from_pretrained(bigname)
+
+#bigmodel = AutoModelForCausalLM.from_pretrained(bigname,  device_map="auto",load_in_8bit=True)
+bigmodel.eval()
+from accelerate import Accelerator
+accelerator = Accelerator()
+
+bigmodel = accelerator.prepare(bigmodel)
+bigmodel.eval()
```
Comment (adstraw) [inline eagle/ge_data/ge_data_all_llama3-1-70Bchat.py]: Is this second eval necessary?

File: eagle/ge_data/ge_data_all_llama3-1chat.py [hunk 4]
```diff
@@ -203,9 +196,12 @@ def writedata(name,data_point):
 
 
 for id,data in enumerate(ds):
+    print(id)
```
Comment (adstraw) [inline eagle/ge_data/ge_data_all_llama3-1chat.py]: delete?

File: eagle/model/cnets.py [hunk 5]
```diff
@@ -600,6 +600,7 @@ def forward(
 
         #hidden_states=self.act(self.fc(torch.cat((inputs_embeds,hidden_states),dim=-1)))
         inputs_embeds=inputs_embeds.to(hidden_states.dtype)
+
```
Comment (adstraw) [inline eagle/model/cnets.py]: unnecessary whitespace

File: eagle/train/main.py [hunk 6]
```diff
@@ -339,6 +337,7 @@ def getkacc(model, data, head, max_length=5):
     model, head, optimizer, train_loader, test_loader = accelerator.prepare(
         model, head, optimizer, train_loader, test_loader
     )
+
```
Comment (adstraw) [inline eagle/train/main.py]: unnecessary whitespace

File: llama-3-1-70B-multi-gpu-train-data-generation.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+nohup /opt/bin/cuda-reserve.py --num-gpus 6 accelerate launch --config_file=multi_gpu_acc.yaml  ge_data_all_llama3-1-70Bchat.py --start=0 --end=68000 --index=0 --gpu_index 0 --outdir ../../train-data-llama-3-1-70b/sharegpt_0_67999_mufp16 > nohup_data_gen_script.out &
```
Comment (adstraw) [inline llama-3-1-70B-multi-gpu-train-data-generation.md]: Is this file needed now that we have doc in the README?

File: eagle/train/main.py [hunk 8]
```diff
@@ -171,9 +171,7 @@ def __getitem__(self, index):
         new_data["target"] = target
         new_data["hidden_state_big"] = hidden_state
         new_data["input_ids"] = input_ids_target
-        # sample = torch.cat((data['xs'],data['xb']))
-        # sample=torch.cat((self.data[index]['x'],self.data[index]['logits']))
-        # label = data['y']
+        print("load ", index, "done")
```
Comment (adstraw) [inline eagle/train/main.py]: delete?

File: eagle/ge_data/allocation.py [hunk 9]
```diff
@@ -55,12 +56,12 @@ def run_command(cmd):
...
+    command = "python ge_data_all_llama3-1chat.py --start={} --end={} --index={} --gpu_index {} --outdir {} --base-model-path".format(
+        start, end, index, gpu_index_str, outdir, args.base_model_path)
     commands.append(command)
 # run_command(commands[0])
 # commands=commands[:1]
 with ThreadPoolExecutor(max_workers=len(commands)) as executor:
     for command in commands:
         executor.submit(run_command, command)
-        print(command)
+        print(command)
```
Comment (adstraw) [inline eagle/ge_data/allocation.py]: unnecessary whitespace diff

**Review by adstraw — CHANGES_REQUESTED** [review]
It looks to me like the 8b and 70b training data scripts have converged with the addition of the --base-model-path argument, which I like.  Can we have just one script?  Left you some review comments for general cleanup

File: llama-3-1-70B-multi-gpu-train-data-generation.md (hunk 7 above)
Comment (ita9naiwa) [reply llama-3-1-70B-multi-gpu-train-data-generation.md]: deleted

File: eagle/train/main.py (hunk 8 above)
Comment (ita9naiwa) [reply eagle/train/main.py]: deleted

File: eagle/ge_data/allocation.py (hunk 9 above)
Comment (ita9naiwa) [reply eagle/ge_data/allocation.py]: fixed

File: eagle/ge_data/ge_data_all_llama3-1chat.py (hunk 4 above)
Comment (ita9naiwa) [reply eagle/ge_data/ge_data_all_llama3-1chat.py]: deleted

File: eagle/ge_data/ge_data_all_llama3-1-70Bchat.py (hunk 3 above)
Comment (ita9naiwa) [reply eagle/ge_data/ge_data_all_llama3-1-70Bchat.py]: I removed 70Bchat completely.

File: eagle/ge_data/ge_data_all_llama3-1chat.py (hunk 2 above)
Comment (ita9naiwa) [reply eagle/ge_data/ge_data_all_llama3-1chat.py]: removed.

File: eagle/ge_data/ge_data_all_llama3-1-70Bchat.py (hunk 1 above)
Comment (ita9naiwa) [reply eagle/ge_data/ge_data_all_llama3-1-70Bchat.py]: I removed 70Bchat completely.

**Pull request merged**